docs/notebooks/measurement-error-mitigation.ipynb

```diff
@@ -0,0 +1,546 @@
+    "from quantuminspire.api import QuantumInspireAPI\n",
+    "from quantuminspire.credentials import get_authentication\n",
+    "from quantuminspire.qiskit import QI\n",
+    "\n",
+    "\n",
+    "def get_qi_calibration_from_qiskit_result(qi_job, api, authentication) -> dict:\n",
+    "    \"\"\" Return calibration data from a QI job \"\"\"\n",
+    "    project = api.get_project(qi_job.job_id())\n",
+    "    # NOTE: due to limited capabilties of the QI framework this is the latest job, perhaps not the actual job\n",
+    "    job = qi_job.get_jobs()[0]\n",
+    "    result = api.get_result_from_job(job[\"id\"])\n",
+    "    c = result[\"calibration\"]\n",
+    "    result = requests.get(c, auth=authentication)\n",
+    "    calibration = result.json()\n",
+    "    return calibration"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "\n",
+    "We will measure a Bell state, first without error mitigation, and then with error mitigation. We perform the calculations on real hardware (or on a simulator when the hardware is not available)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def spin_2_noise_model(p: float = 0.01):\n",
+    "    \"\"\" Define noise model representation for Spin-2 \"\"\"\n",
+    "    noise_model = NoiseModel()\n",
+    "    error_gate = pauli_error([(\"X\", p), (\"I\", 1 - p)])\n",
+    "\n",
+    "    noise_model.add_all_qubit_quantum_error(error_gate, [\"u1\", \"u2\", \"u3\", \"rx\", \"ry\", \"x\", \"y\"])\n",
+    "\n",
+    "    read_err = ReadoutError([[0.9, 0.1], [0.2, 0.8]])\n",
+    "    noise_model.add_readout_error(read_err, [0])\n",
+    "    read_err = ReadoutError([[0.7, 0.3], [0.25, 0.75]])\n",
+    "    noise_model.add_readout_error(read_err, [1])\n",
+    "    return noise_model\n",
+    "\n",
+    "\n",
+    "noise_model_spin2 = spin_2_noise_model()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set up connection to QI"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We expect the user to have set up the token authentication as described [here](https://github.com/qutech-delft/quantuminspire#configure-your-token-credentials-for-quantum-inspire) in order to be able to access the Spin-2 backend in the following cells."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Enter email:\n",
+      "example@email.com\n",
+      "Enter password\n",
+      "········\n",
+      "QI connection not available, fall-back to simulation.\n"
+     ]
+    }
+   ],
+   "source": [
+    "try:\n",
+    "    authentication = get_authentication()\n",
+    "    qi_api = QuantumInspireAPI(authentication=authentication)\n",
```

Comment:
Actually here and in the next line you are authenticating (twice). The reason is to get the api because it is needed elsewhere in the code. The latest version of quantuminspire (1.5.0 released a day ago) has a method to get the api from QI. That can be used instead. Where you need api you can do `QI.get_api()`
Consequences: this line can be deleted